account_report_invoice_paid/views/report_invoice_templates.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//div[hasclass('page')]/*[1]" position="replace">
+            <div t-if="o.state != 'paid'">
+                <h2>
+                    <span t-if="o.type == 'out_invoice' and o.state in ('open', 'in_payment')">Invoice</span>
+                    <span t-if="o.type == 'out_invoice' and o.state == 'draft'">Draft Invoice</span>
+                    <span t-if="o.type == 'out_invoice' and o.state == 'cancel'">Cancelled Invoice</span>
+                    <span t-if="o.type == 'out_refund'">Credit Note</span>
+                    <span t-if="o.type == 'in_refund'">Vendor Credit Note</span>
+                    <span t-if="o.type == 'in_invoice'">Vendor Bill</span>
+                    <span t-field="o.number" />
+                </h2>
+            </div>
+            <div t-if="o.state == 'paid'">
+                <h2>
+                    <span>Paid Invoice</span>
+                </h2>
+            </div>
+        </xpath>
```

Comment:
This xpath expr is too wide. It could raise conflicts with other modules.
Let's say there's another module that adds a new element at the begining of the page, it would break the layout.

Also according to this it'll print "Paid Invoice" for Vendor invoices aswell. And we only want it for sale invoices

I haven't tested this, but can you change it to something like this?

```suggestion
        <span t-if="o.type == 'out_invoice' and o.state in ('open', 'in_payment', 'paid')" position="replace">
            <span t-if="o.type == 'out_invoice' and o.state in ('open', 'in_payment')">Invoice</span>
            <span t-if="o.type == 'out_invoice' and o.state == 'paid'">Paid Invoice</span>
        </span>
```


I'm not sure if we can target elements like this. But please test it.
if it doesn't work we can do something like this:

```suggestion
        <xpath expr="//div[hasclass('page')]/h2/span[1]" position="replace">
            <span t-if="o.type == 'out_invoice' and o.state in ('open', 'in_payment')">Invoice</span>
            <span t-if="o.type == 'out_invoice' and o.state == 'paid'">Paid Invoice</span>
        </xpath>
```

But I think the first suggestion is more explicit, so if it works it's definitely better